mitodl/mitxpro
# feat: How you will learn and B2B section added for external courses

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/6064
https://github.com/mitodl/hq/issues/6065

### Description (What does it do?)
It creates a new page type `CommonExternalCoursePage` to act as a folder and hold static pages that needs to be auto created when an ExternalCoursePage is created

### How can this be tested?
1. Login as admin and go to CMS site of xPro
2. Create a page under CourseIndexPage named "Common External Course Page"
3. Add Childs to "CommonExternalCoursePage" e.g; `LearningTechniqueExternalCoursePage` and `ForTeamsExternalCoursePage` (Platform is optional)
4. Create ExternalCoursePage
5. Newly created external course page should have copies of child that were created in `CommonExternalCoursePage` in step 3

### Additional Context
1. If there is no child page of "CommonExternalCoursePage" - No copies will be created for newly created external course page
2. If there are child pages of "CommonExternalCoursePage" with specific platform - Copies of these child pages will be created only for that platform's external course pages
3. If there are child pages of "CommonExternalCoursePage" with and without platform - Copies of these child pages will be created of the same platform or from without platform pages
